File: solutions/c/kindergarten-garden/1/kindergarten_garden.c
#include "kindergarten_garden.h"

plants_t plants(const char *diagram, const char *student)
{
    const char *it = diagram;
    while (*(it++) != '\n');

    plants_t p;
    int index = (*student - 'A');
    p.plants[0] = diagram[2 * index];
    p.plants[1] = diagram[2 * index +1];
    p.plants[2] = it[2 * index];
    p.plants[3] = it[2 * index + 1];
    return p;
}

File: solutions/csharp/kindergarten-garden/1/KindergartenGarden.cs
using System;
using System.Collections.Generic;

public enum Plant
{
    Violets,
    Radishes,
    Clover,
    Grass
}

public class KindergartenGarden
{
    private string[] _garden;
    private char[] _key = { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l' };
    public KindergartenGarden(string diagram)
    {
        _garden = diagram.Split("\n");
    }

    private Plant GetPlant(char value)
    {
        switch (value)
        {
            case 'V': return Plant.Violets;
            case 'R': return Plant.Radishes;
            case 'C': return Plant.Clover;
            case 'G': return Plant.Grass;
        }
        throw new Exception("I don't know of this plant");
    }
    public IEnumerable<Plant> Plants(string student)
    {
        var ret = new List<Plant>();
        var studentIndex = Array.IndexOf(_key, student.ToLower()[0]) * 2;

        ret.Add(GetPlant(_garden[0][studentIndex]));
        ret.Add(GetPlant(_garden[0][studentIndex + 1]));
        ret.Add(GetPlant(_garden[1][studentIndex]));
        ret.Add(GetPlant(_garden[1][studentIndex + 1]));

        return ret;
    }
}